File: BuzzStats.WebApi/DTOs/RecentActivity.cs
using System;

namespace BuzzStats.WebApi.DTOs
{
    /// <summary>
    /// Recent activity data object.
    /// </summary>
    public class RecentActivity
    {
        public int StoryId { get; set; }
        public int CommentId { get; set; }
        public string Title { get; set; }
        public DateTime CreatedAt { get; set; }

        public string StoryUsername { get; set; }
        public DateTime StoryCreatedAt { get; set; }

        public string StoryVoteUsername { get; set; }

        public string CommentUsername { get; set; }
        public DateTime CommentCreatedAt { get; set; }
    }
}

File: BuzzStats.WebApi/scripts/RecentActivity.js
import React from 'react';
import {toAgo} from './date';

function recentActivityToText(recentActivity) {
    if (!recentActivity) {
        return 'N/A';
    }

    if (recentActivity.storyVoteUsername) {
        return 'Story ' + recentActivity.title + ' voted by user ' + recentActivity.storyVoteUsername;
    }

    if (recentActivity.commentUsername) {
        return 'New comment on story ' + recentActivity.title + ' by user ' + recentActivity.commentUsername +
            ' (' + toAgo(recentActivity.createdAt) + ')';
    }

    return 'New story ' + recentActivity.title + ' by user ' + recentActivity.storyUsername +
        ' (' + toAgo(recentActivity.createdAt) + ')';
}

export default class extends React.Component {
    render() {
        return <ol className="recent-activity">
            {this.props.data.map(d =>
                <li>{recentActivityToText(d)}</li>
            )}
        </ol>;
    }
}
